CRM/Civisocial/OAuthProvider/Facebook.php

```diff
@@ -173,14 +132,24 @@ public function handleCallback() {
    * @return bool
    */
   public function isAuthorized() {
-    if ($this->token && isset($this->userProfile)) {
+    if ($this->token && !empty($this->userProfile)) {
```

Comment:
@siddharthg Well, I will have to document that but the idea is that `isAuthorized()` is called before any other operations. General steps would be:

1. Retrieve the token (from session/database or login dialog) and set it to an OAuthProvider object.
2. Token could be expired or access might have been revoked. And the standard way for many OAuth providers are to make a request for profile. That's what `isAuthorized()` does. It groups the token verification and profile fetching operation.